compiler/util/canonicalize_affine.py

```diff
@@ -0,0 +1,121 @@
+from xdsl.ir.affine import (
+    AffineBinaryOpExpr,
+    AffineBinaryOpKind,
+    AffineConstantExpr,
+    AffineDimExpr,
+    AffineExpr,
+    AffineMap,
+)
+
+
+def get_dim(expr: AffineExpr) -> int | None:
+    # get the dimension of this expression. If d0 appears, return 0.
+    # If d1 appears, return 1. This will fail for nested additions in
+    # which multiple dimension expressions appear. This happens quite
+    # often, but can only be solved with a full flattening of the affine map.
+    # this is todo
```

Comment:
```suggestion
    """
    get the dimension of this expression. If d0 appears, return 0.
    If d1 appears, return 1. This will fail for nested additions in
    which multiple dimension expressions appear. This happens quite
    often, but can only be solved with a full flattening of the affine map.
    this is todo
    """
```